src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.TestClasses.Collections.cs

```diff
@@ -388,5 +389,78 @@ public class Element
         {
             public string Type { get; set; }
         }
+
+        public class OptionsWithStructs
+        {
+            public StringValues StringValues { get; set; }
+
+            public CollectionStructExplicit CollectionStructExplicit { get; set; } = new();
+
+            public DictionaryStructExplicit DictionaryStructExplicit { get; set; } = new();
+        }
+
+        public struct CollectionStructExplicit : ICollection<string>
+        {
+            public CollectionStructExplicit() {}
+
+            ICollection<string> _collection = new List<string>();
+
+            int ICollection<string>.Count => _collection.Count;
+
+            bool ICollection<string>.IsReadOnly => _collection.IsReadOnly;
+
+            void ICollection<string>.Add(string item) => _collection.Add(item);
+
+            void ICollection<string>.Clear() => _collection.Clear();
+
+            bool ICollection<string>.Contains(string item) => _collection.Contains(item);
+
+            void ICollection<string>.CopyTo(string[] array, int arrayIndex) => _collection.CopyTo(array, arrayIndex);
+
+            IEnumerator<string> IEnumerable<string>.GetEnumerator() => _collection.GetEnumerator();
+
+            IEnumerator IEnumerable.GetEnumerator() => ((IEnumerable)_collection).GetEnumerator();
+
+            bool ICollection<string>.Remove(string item) => _collection.Remove(item);
+        }
+
+        public struct DictionaryStructExplicit : IDictionary<string, string>
```

Comment:
> public struct DictionaryStructExplicit : IDictionary<string, string> [](http://example.com/codeflow?start=7&length=69)

Can we add test for struct implmenting IReadOnlyDictionary and IReadOnlyCollection?

Reply:
I can - both will be "unsupported" since they don't implement the writable interfaces.  The downcast check only happens when the bound type itself is IRO interface - in which case it's always safe to test cast the interface object.

Reply:
I added those tests to demonstrate that they do nothing.

Reply:
LGTM